project_cost_report/static/src/js/report_action.js

```diff
@@ -0,0 +1,386 @@
+        this.updateHtml();
+        this.updateFoldButtonVisibility();
+    },
+    /**
+     * Download the PDF version of the report.
+     */
+    downloadPDF(){
+        framework.blockUI();
+        session.get_file({
+            url: "/web/project_cost_report/" + this.projectId,
+            data: {report_context: JSON.stringify(this.reportContext)},
+            complete: framework.unblockUI,
+            error: crash_manager.rpc_error.bind(crash_manager),
+        });
+    },
+    /**
+     * Fold all report categories.
+     */
+    fold(){
+        this.reportContext.unfolded_categories = {
+            'product': [],
+            'time': [],
+            'outsourcing': [],
+        };
+        this.unfolded = false;
+        this.refresh();
+    },
+    /**
+     * Unfold all report categories.
+     */
+    async unfold(){
+        this.reportContext.unfolded_categories = await this._rpc({
+            model: 'project.cost.report',
+            method: 'get_foldable_categories',
+            args: [this.projectId],
+        });
+        this.unfolded = true;
+        this.refresh();
+    },
+    /**
+     * Fold a single category.
+     *
+     * @param {Number} categoryId - the ID of the category.
+     * @param {String} sectionName - the name of the section.
+     */
+    foldCategory(categoryId, sectionName){
+        this.reportContext.unfolded_categories[sectionName] = (
+            this.reportContext.unfolded_categories[sectionName].filter((c) => c != categoryId)
+        );
+        this.unfolded = false;
+        this.refresh();
+    },
+    /**
+     * Unfold a single category.
+     *
+     * @param {Number} categoryId - the ID of the category.
+     * @param {String} sectionName - the name of the section.
+     */
+    unfoldCategory(categoryId, sectionName){
+        this.reportContext.unfolded_categories[sectionName].push(categoryId);
+        this.refresh();
+    },
+    /**
+     * Handle a click on the label of a category.
+     *
+     * The category is toggled (folded or unfolded).
+     *
+     * @param {Number} categoryId - the ID of the category.
+     * @param {String} sectionName - the name of the section.
+     */
+    categoryClicked(categoryId, sectionName){
+        event.preventDefault();
+        var unfoldedIds = this.reportContext.unfolded_categories[sectionName];
+        var isUnfolded = unfoldedIds.indexOf(categoryId) !== -1;
+        if(isUnfolded){
+            this.foldCategory(categoryId, sectionName);
+        }
+        else{
+            this.unfoldCategory(categoryId, sectionName);
+        }
+    },
+    /**
+     * Handle a click on an analytic line.
+     *
+     * The form view of the analytic is opened.
+     *
+     * @param {jQuery.Event} - the click event
+     */
+    analyticLineClicked(event){
+        event.preventDefault();
+        var lineId = getRecordIdFromEvent(event);
+        this.do_action({
+            res_model: "account.analytic.line",
+            res_id: lineId,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Identifier 'res_id' is not in camel case. (camelcase)](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=3675949)